src/internal_events/common.rs

```diff
@@ -157,47 +157,24 @@ impl<E: std::fmt::Display> InternalEvent for SinkRequestBuildError<E> {
 }
 
 #[derive(Debug)]
-pub struct SinkRetryReasonError<'a> {
-    pub message: &'static str,
-    pub reason: Cow<'a, str>,
-}
-
-impl InternalEvent for SinkRetryReasonError<'_> {
-    fn emit(self) {
-        error!(
-            message = %self.message,
-            reason = %self.reason,
-            error_type = error_type::REQUEST_FAILED,
-            stage = error_stage::SENDING,
-            rate_limit_secs = 10,
-        );
-        counter!(
-            "component_errors_total", 1,
-            "error_type" => error_type::REQUEST_FAILED,
-            "stage" => error_stage::SENDING,
-        );
-    }
-}
-
-#[derive(Debug)]
-pub struct SinkRetryError<E> {
+pub struct SinkSendError<E> {
     pub message: &'static str,
     pub error: E,
 }
 
-impl<E: std::fmt::Display> InternalEvent for SinkRetryError<E> {
+impl<E: std::fmt::Display> InternalEvent for SinkSendError<E> {
     fn emit(self) {
         error!(
             message = %self.message,
             error = %self.error,
-            error_type = error_type::ENCODER_FAILED,
-            stage = error_stage::PROCESSING,
+            error_type = error_type::COMMAND_FAILED,
```

Comment:
I don't think `COMMAND_FAILED` is right, out of the list `REQUEST_FAILED` seems more correct.

Reply:
Ah, good call, agreed.